src/Fmt_ast.ml

```diff
@@ -1380,25 +1380,19 @@ and fmt_expression c ?(box = true) ?epi ?eol ?parens ?ext
                         $ fmt_fun_args c xargs $ fmt "@ ->" ) )
                   $ fmt "@ " $ fmt_expression c xbody )) ))
   | Pexp_apply
-      ( { pexp_desc= Pexp_ident {txt= Ldot (Lident "Array", "get")}
-        ; pexp_attributes= [] }
-      , [(Nolabel, s); (Nolabel, i)] ) ->
-      fmt_index_op c ctx ~parens index_op_array s i
-  | Pexp_apply
-      ( { pexp_desc= Pexp_ident {txt= Ldot (Lident "String", "get")}
-        ; pexp_attributes= [] }
-      , [(Nolabel, s); (Nolabel, i)] ) ->
-      fmt_index_op c ctx ~parens index_op_string s i
+      ( {pexp_desc= Pexp_ident {txt}; pexp_attributes= []}
+      , (Nolabel, s) :: indices )
+    when Option.is_some (index_op_get_sugar txt indices) ->
+      let op, indices = Option.value_exn (index_op_get_sugar txt indices) in
```

Comment:
It would be nice to avoid the repetition between the when clause and the body, but I don't know a composable way to write it.